### PR TITLE
Calculate search score using streams.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- In the preference search, calculate the preference score using the Streams API.

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java [hunk 1]
```diff
@@ -72,39 +72,22 @@ static class FuzzySearchSpecificDTO {
...
-            if (countOfAttributesWithScore != 0) {
-                this.score = attributeScoreSum / countOfAttributesWithScore;
-            } else {
-                this.score = 0;
-            }
+            this.score = WEIGHT_MAP.entrySet().stream().mapToDouble(entry -> {
+                final String valueToProcess = entry.getKey().apply(item);
+                return valueToProcess.isEmpty()
+                        ? -1 : FUZZY_SCORE.fuzzyScore(valueToProcess, keyword) * entry.getValue();
+            }).filter(value -> value != -1).average().orElse(0);
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java:83]: Wouldn't this be better? Also keep the formatting
```suggestion
            this.score = WEIGHT_MAP.entrySet().stream()
                    .map(entry -> new Pair<>(entry.getKey().apply(item), entry.getValue()))
                    .filter(pair -> !pair.first.isEmpty())
                    .mapToDouble(pair -> FUZZY_SCORE.fuzzyScore(pair.first, keyword) * pair.second)
                    .average()
                    .orElse(0);
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Please don't chain streams logical operations one after the other on the same line... It makes everything much less understandable.

File: app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java [hunk 2]
```diff
@@ -72,39 +73,23 @@ static class FuzzySearchSpecificDTO {
         );
 
         private final PreferenceSearchItem item;
-        private final float score;
+        private final double score;
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java:76]: Wait, why did you convert it to `double`?

Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceFuzzySearchFunction.java:76]: > Wait, why did you convert it to `double`?

The streams API doesn't support mapping to `float` directly. It works as it did before, though.

**Comment by TacoTheDank** [comment]
There are four commits... could this be smooshed into one (seeing as this PR is small)?

**Review by Stypox — APPROVED** [review]

**Pull request merged**